store/store.go

```diff
@@ -379,25 +388,66 @@ func (s *store) IsBanned(addr crypto.Address) bool {
 	return s.config.BannedAddrs[addr]
 }
 
+func (s *store) Prune(resultFunc func(pruned, skipped, pruningHeight uint32)) error {
+	cert := s.lastCertificate()
+
+	// at genesis block
+	if cert == nil {
+		return nil
+	}
+
+	retentionBlocks := uint32(s.config.RetentionDays * totalBlockInDay)
+
+	if cert.Height() < retentionBlocks {
+		return nil
+	}
+
+	pruningHeight := cert.Height() - retentionBlocks
+
+	// start pruning blocks and transactions
+	for i := pruningHeight; i >= 1; i-- {
+		deleted, err := s.pruneBlock(i)
+		if err != nil {
+			return err
+		}
+
+		if deleted {
+			resultFunc(1, 0, i)
+
+			continue
+		}
+
+		resultFunc(0, 1, i)
+	}
+
+	return nil
+}
+
 func (s *store) pruneBlock(blockHeight uint32) (bool, error) { //nolint
```

Comment:
You can remove `//nolint` from here.

Reply:
This function needs to unit test.

Reply:
I know, I was talking about removing the comment...